### ISSUE TITLE
`needless_pass_by_value` shouldn't trigger custom types with generics

### ISSUE THREAD

**Issue by c410-f3r** [rust-lang/rust-clippy#10077]
A generic type can be a thing that implements `Copy`, which makes the lint suggest `#[derive(Copy)]` for the underlying type.

```rust
#![deny(clippy::needless_pass_by_value)]

struct _Foo<T: AsRef<[u8]>>(T);

type _FooRef<'a> = _Foo<&'a [u8]>;
// `Vec` does not implement `Copy`
type _FooVec = _Foo<Vec<u8>>;

fn _foo(elem: _FooRef<'_>) -> usize {
    elem.0.len()
}
```

**Comment by c410-f3r** [comment]
Closing due to the lack of activity.
